dulwich/objects.py

```diff
@@ -1506,6 +1506,13 @@ def sign(self, keyid: Optional[str] = None):
                     self.as_raw_string(), mode=gpg.constants.sig.mode.DETACH
                 )
 
+    def without_gpgsig(self) -> bytes:
```

Comment:
the name without_gpgsig suggests to me that this will return a Commit rather than bytes.

Reply:
good point. Some brainstorming here:

1. `commit.bytes_without_gpgsig()`
2. `commit.as_raw_without_gpgsig()`

And `gpgsig` could be shortened to `sig` which has some logic to it because the signature can be with ssh instead of gpg.

Reply:
I like the idea of `commit._____without_sig()` except that a user might be unsure whether the `sig` removed is the same contents as what the property `gpgsig` returns.

Reply:
does it need to return the bytes rather than the actual object? The caller could always call .as_raw_string() if they wanted to.

Dropping the "gpg" bit makes sense to me.

Reply:
And there's `def __bytes__(self): -> bytes` too.

Let's image a user wants to check the commit signature and any public key is ok.
```
check_signature(commit.raw_without_sig(), commit.gpgsig)
```
v.s.
```
check_signature(bytes(commit.without_sig()), commit.gpgsig)
```
or equiv
```
check_signature(commit.without_sig().as_raw_string(), commit.gpgsig)
```

Ya, returning the object and doing `bytes(commit.without_sig())` looks like a winner to me! What do ya think?

Reply:
looks good to me too :)

Reply:
One heads up on this return-a-Commit approach. If there is a Commit.without_sig() it makes sense there should also be a Tag.without_sig(). And the current plan is the equivalent signed cryptographic message for a tag is accessed by doing:
```
bytes(tag.without_sig())
```
which to a human reader is intuitive.

However that will be an inefficient way to accomplish what is currently done in the code as:
```
self.as_raw_string()[: -len(self._signature)]
```
where `self` is a tag object.

But I'm thinking this is OK since this is Python, not C or Rust. But my old brain can't help but cringe a bit at how much CPU and RAM is used by friendly readable Python :nerd_face:.